import_service/api_gateway.js

```diff
@@ -18,12 +18,24 @@ class ApiGateway extends Stack {
       import_service_fn = aws_lambda,
     } = props;
 
+    const basicAuthorizerLambda = aws_lambda.Function.fromFunctionName(this, 'authFunction', 'AuthFunction');
+
     const api = new aws_apigateway.RestApi(this, 'ImportApi', {
         restApiName: 'Import Service'
     });
 
+    const authorizer = new aws_apigateway.TokenAuthorizer(this, 'BasicAuthorizer', {
+      identitySource: 'method.request.header.Authorization',
+      handler: basicAuthorizerLambda
+    });
+
     const import_resource = api.root.addResource('import');
-    import_resource.addMethod('GET', new aws_apigateway.LambdaIntegration(import_service_fn));
+    import_resource.addMethod('GET', new aws_apigateway.LambdaIntegration(import_service_fn),
```

Comment:
This piece of code possibly (I can't give 100%, but still) may fix FE for you. It seems to fail on CORS. There may be some more stuff to add - but it's worth to try.

```
importResource.addMethod('OPTIONS', new apigateway.MockIntegration({
      integrationResponses: [
        {
          statusCode: '200',
          responseParameters: {
            'method.response.header.Access-Control-Allow-Headers': "'Content-Type,X-Amz-Date,Authorization,X-Api-Key,X-Amz-Security-Token'",
            'method.response.header.Access-Control-Allow-Origin': "'*'",
            'method.response.header.Access-Control-Allow-Methods': "'GET,POST,OPTIONS'",
            'method.response.header.Access-Control-Max-Age': "'1728000'",
          },
          responseTemplates: {
            'application/json': '{"statusCode": 200}'
          },
        },
      ],
      passthroughBehavior: apigateway.PassthroughBehavior.NEVER,
      requestTemplates: {
        'application/json': '{"statusCode": 200}'
      },
    }), {
      methodResponses: [
        {
          statusCode: '200',
          responseParameters: {
            'method.response.header.Access-Control-Allow-Headers': true,
            'method.response.header.Access-Control-Allow-Methods': true,
            'method.response.header.Access-Control-Allow-Origin': true,
            'method.response.header.Access-Control-Max-Age': true,
          },
        },
      ],
    });
```